### PR TITLE
Added Tailwind CSS to Collage Section

### PR DESCRIPTION
**Pull request by tanujbordikar**
Solves Issue #118 

- Added Tailwind CSS with dynamic theme configuration

https://github.com/dear-digital/shopify-theme-skeleton/assets/91179998/947315dd-1054-4c73-853c-83d433f51dd6

### CODE REVIEW

File: sections/dev__collage.liquid [hunk 1]
```diff
@@ -2,59 +2,71 @@
 {% assign section_classes = 'section-' | append: section_id | append: '-padding' %}
 
 <div class="color-{{ section.settings.color_scheme }} gradient isolate">
```
Comment (mihir-bombay-studio) [inline sections/dev__collage.liquid:4]: no hardcoded style classes

File: sections/dev__collage.liquid [hunk 2]
```diff
@@ -2,59 +2,71 @@
...
-  <div class="page-width {{ section_classes }}">
-    {% if section.settings.heading != blank %}
-      <h2 class="collage-wrapper-title inline-richtext {{ section.settings.heading_size }}">
+  {% if section.settings.heading != blank %}
+    <div class="collage-wrapper-title {{ section.settings.collage-wrapper-title }}">
+      <div class="inline-richtext">
```
Comment (mihir-bombay-studio) [inline sections/dev__collage.liquid:7]: missing styling class

File: sections/dev__collage.liquid [hunk 3]
```diff
@@ -2,59 +2,71 @@
...
         {{ section.settings.heading }}
-      </h2>
-    {% endif %}
-    <div class="collage{% if section.settings.mobile_layout == 'collage' %} collage--mobile{% endif %}">
+      </div>
+    </div>
+  {% endif %}
+  <div class="page-width {{ section.settings.page-width }}">
```
Comment (mihir-bombay-studio) [inline sections/dev__collage.liquid:12]: All style classes id's should have a styles or classes keyword appended or prepended section.settings.page-width will become
section.settings.page-width-styles or section.settings.page-width-classes

File: sections/dev__collage.liquid [hunk 4]
```diff
@@ -2,59 +2,71 @@
...
+    <div class="collage {{ section.settings.collage }} {% if section.settings.mobile_layout == 'collage' %} collage--mobile{% endif %}">
```
Comment (mihir-bombay-studio) [inline sections/dev__collage.liquid:13]: mobile layout settings can also be done with tailwind classes so no need for extra mobile filters

File: sections/dev__collage.liquid [hunk 5]
```diff
@@ -2,59 +2,71 @@
...
-                    <h3 class="product__title">{{ block.settings.product.title }}</h3>
-                    <p class="product__description">{{ block.settings.product.description }}</p>
-                    <p class="product__price">Price: {{ block.settings.product.price | money }}</p>
-                    <p class="product__sku">SKU: {{ block.settings.product.sku }}</p>
-                    <!-- Add more product details as needed -->
-                    <a href="{{ block.settings.product.url }}" class="btn btn-primary">Shop Now</a>
+        <div class="items-container">
+          {%- assign image_url_width = section.settings.image_url_width -%}
+          {%- assign image_tag_width = section.settings.image_tag_width -%}
+          <div class="collage__item {{ section.settings.collage__item }} collage__item--{{ block.type }} collage__item--{{ section.settings.desktop_layout }}">
```
Comment (mihir-bombay-studio) [inline sections/dev__collage.liquid:18]: all block level styling should be at block level only. also no need to have desktop layout as all styling is possible from tailwind classes

File: sections/dev__collage.liquid [hunk 6]
```diff
@@ -2,59 +2,71 @@
...
+            {% case block.type %}
+              {% when 'image' %}
+                <div class="collage-card {{ section.settings.collage-card }}">
+                  <div class="media media--transparent ratio">
```
Comment (mihir-bombay-studio) [inline sections/dev__collage.liquid:22]: missing tailwind classes

**Review by mihir-bombay-studio — CHANGES_REQUESTED** [review]
@tanujbordikar I've added some comments can you check them and correct wherever required in this file and also on your other PR's

**Comment by tanujbordikar** [comment]
Hey @mihir-bombay-studio Sir, I have made the necessary changes as you said! Can you please review the new code.

File: sections/dev__collage.liquid [hunk 7]
```diff
@@ -1,60 +1,62 @@
-{% assign section_id = section.id %}
-{% assign section_classes = 'section-' | append: section_id | append: '-padding' %}
-
-<div class="color-{{ section.settings.color_scheme }} gradient isolate">
-  <div class="page-width {{ section_classes }}">
-    {% if section.settings.heading != blank %}
-      <h2 class="collage-wrapper-title inline-richtext {{ section.settings.heading_size }}">
+  <div class="color-{{ section.settings.color_scheme }} gradient isolate">
```
Comment (mihir-bombay-studio) [inline sections/dev__collage.liquid:1]: This element still has hardcoded classes @tanujbordikar

File: sections/dev__collage.liquid [hunk 8]
```diff
@@ -1,60 +1,62 @@
...
+                {% when 'collection' %}
+                <div class="collection-card {{ block.settings.collection-card }}">
+                  <div class="collection">
+                    {%- if block.settings.collection != blank -%}
+                      {{ block.settings.collection | image_url: width: image_url_width | image_tag: width: image_tag_width }}
+                    {% else %}
+                      {{ 'image' | placeholder_svg_tag }}
+                    {% endif %}
+                    <h3 class="collection__title">{{ block.settings.collection.title }}</h3>
+                    <p class="collection__description text-base">{{ block.settings.collection.description }}</p>
```
Comment (mihir-bombay-studio) [inline sections/dev__collage.liquid:45]: These elements don't have tailwind schema

**Review by mihir-bombay-studio — APPROVED** [review]

**Pull request merged**